File: Algorithms/PointsCover/PointsCover.cpp
#include <vector>
#include <iostream>
#include <bits/stdc++.h>

using Segment = std::pair<int, int>;

// Comprator
bool sortBySecondValue(const std::pair<int,int> &firstElement, const std::pair<int,int> &secondElement) {
    return (firstElement.second < secondElement.second);
}

// Points Covering
std::vector <int> getCoveringSet(std::vector <Segment> segments) {
  std::vector <int> points;

  std::sort(segments.begin(), segments.end(), sortBySecondValue);

  for (auto segment : segments) {
    if (points.empty() or (segment.first > points.back())) {
      points.push_back(segment.second);
    }
  }

  return points;
}

int main(void) {
  int segmentsCount;
  std::cin >> segmentsCount;
  std::vector <Segment> segments(segmentsCount);
  for (auto &segment:segments) {
    std::cin >> segment.first >> segment.second;
  }

  auto points = getCoveringSet(std::move(segments));
  std::cout << points.size() << std::endl;
  for (auto point:points) {
    std::cout << point << '\t';
  }
  std::cout << std::endl;
}

File: Algorithms/PointsCover/PointsCover.py
def get_covering_set(segments):
    points = []
    segments.sort(key=lambda value: value[1])

    for segment in segments:
        if ((not points) or (segment[0] > points[-1])):
            points.append(segment[1])

    return points


def main():
    segments_count = int(input())
    segments = []
    for _ in range(segments_count):
        first, second = input().split()
        segments.append((int(first), int(second)))
    
    points = get_covering_set(segments)
    print(len(points))

    for idx, _ in enumerate(points):
        if idx < len(points) - 1:
            print(points[idx], end=' ')
        else:
            print(points[idx])


if __name__ == '__main__':
    main()
